lib/platforms/ios.dart

```diff
@@ -107,34 +107,100 @@ void _setIOSPackageName(dynamic packageName) {
     }
 
     final iosProjectString = iosProjectFile.readAsStringSync();
-    final newBundleIDIOSProjectString = iosProjectString
-        // Replaces old bundle id from
-        // `PRODUCT_BUNDLE_IDENTIFIER = {{BUNDLE_ID}};`
-        .replaceAll(
-          RegExp(
-            r'PRODUCT_BUNDLE_IDENTIFIER = ([A-Za-z0-9.-_]+)(?<!\.RunnerTests);',
-          ),
-          'PRODUCT_BUNDLE_IDENTIFIER = $packageName;',
-        )
-        // Replaces old bundle id from
-        // `PRODUCT_BUNDLE_IDENTIFIER = {{BUNDLE_ID}}.RunnerTests;`
-        .replaceAll(
-          RegExp('PRODUCT_BUNDLE_IDENTIFIER = (.*?).RunnerTests;'),
-          'PRODUCT_BUNDLE_IDENTIFIER = $packageName.RunnerTests;',
-        )
-        // Removes old bundle id from
-        // `PRODUCT_BUNDLE_IDENTIFIER = "{{BUNDLE_ID}}.{{EXTENSION_NAME}}";`
-        .replaceAllMapped(
-      RegExp(
-        r'PRODUCT_BUNDLE_IDENTIFIER = "([A-Za-z0-9.-_]+)\.([A-Za-z0-9.-_]+)";',
-      ),
-      (match) {
-        final extensionName = match.group(2);
-        return 'PRODUCT_BUNDLE_IDENTIFIER = "$packageName.$extensionName";';
-      },
+
+    // Extract all bundle identifiers, accounting for both quoted and unquoted formats
+    final bundleIdRegex = RegExp(
+      r'PRODUCT_BUNDLE_IDENTIFIER = "?([^";]+)"?;',
```

Comment:
Should use only the allowed characters to match group. i.e. ``r'PRODUCT_BUNDLE_IDENTIFIER = "?([A-Za-z0-9.-_]+)"?;'`